pkg/http/server.go

```diff
@@ -69,6 +70,17 @@ func (s *server) setupListener(opts Opts) error {
 		return nil
 	}
 
+	listeners, err := activation.Listeners()
+	if err == nil && len(listeners) > 0 {
```

Comment:
What if there is an error here? Seems like it's not being reported? 

Can we reverse this a bit to do the `if err != nil` idiomatic check and then go on to the listeners check

Reply:
I tried out https://darkcoding.net/software/systemd-socket-activation-in-go/ lately and that worked perfecly.
Maybe we should check if the path is systemd:///dev/fd/<number> and just listen on that FD instead.

That way we can skip the whole import of activation.

Reply:
Actually, if we just let the user send an argument with which FD we should use, the code becomes much simpler.

Reply:
Done.